pkg/operator/ceph/object/user/controller.go

```diff
@@ -434,7 +502,7 @@ func generateUserCaps(user admin.User) string {
 	return caps
 }
 
-func generateUserConfig(user *cephv1.CephObjectStoreUser) *admin.User {
+func (r *ReconcileObjectStoreUser) generateUserConfig(user *cephv1.CephObjectStoreUser) (*admin.User, error) {
```

Comment:
This function can be unit tested more easily if the keys from secrets are passed in, rather than read from a k8s object. I'd prefer to keep k8s object reading outside of this func

Reply:
`generateUserKeySpec()` has been split out of this func.